k8s-plugin/src/test/kotlin/com/amazon/spinnaker/keel/tests/K8sResourceHandlerTest.kt

```diff
@@ -561,6 +593,29 @@ internal class K8sResourceHandlerTest : JUnit5Minutests {
             }
         }
 
+        context("when spec with items are used") {
+            val current = yamlMapper.readValue(listYaml, K8sResourceSpec::class.java)
+            val currentSpec = resource(
+                kind = K8S_RESOURCE_SPEC_V1.kind,
+                spec = current
+            )
+
+            test("each item should have the plugin labels") {
+                runBlocking {
+                    val spec = desired(currentSpec)
+                    expectThat(spec.items?.size).isEqualTo(2)
+                    spec.items?.map {
+                        val labels = it.metadata[LABELS] as MutableMap<String, String>
```

Comment:
do we need to verify a null check too? for `Deployment`?

Reply:
Nope. This is to ensure that the existing labels are preserved. If you look at the test yaml above, only the `Service` kind has an existing label pair.